op/k8s/config.go

```diff
@@ -41,128 +44,43 @@ func kubeletKubeconfig(cluster string, n *cke.Node, caPath, certPath, keyPath st
-
-	ContainerLogMaxSize  string `json:"containerLogMaxSize,omitempty"`
-	ContainerLogMaxFiles int32  `json:"containerLogMaxFiles,omitempty"`
-}
-
-func newKubeletConfiguration(cert, key, ca, domain, logSize string, logFiles int32, allowSwap bool) KubeletConfiguration {
-	return KubeletConfiguration{
-		APIVersion:            "kubelet.config.k8s.io/v1beta1",
-		Kind:                  "KubeletConfiguration",
-		ReadOnlyPort:          0,
-		TLSCertFile:           cert,
-		TLSPrivateKeyFile:     key,
-		Authentication:        KubeletAuthentication{ClientCAFile: ca},
-		Authorization:         kubeletAuthorization{Mode: "Webhook"},
+func newKubeletConfiguration(cert, key, ca, domain, logSize string, logFiles int32, allowSwap bool) kubeletv1beta1.KubeletConfiguration {
+	return kubeletv1beta1.KubeletConfiguration{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "kubelet.config.k8s.io/v1beta1",
+			Kind:       "KubeletConfiguration",
+		},
+		ReadOnlyPort:      0,
+		TLSCertFile:       cert,
+		TLSPrivateKeyFile: key,
+		Authentication: kubeletv1beta1.KubeletAuthentication{
+			X509:    kubeletv1beta1.KubeletX509Authentication{ClientCAFile: ca},
+			Webhook: kubeletv1beta1.KubeletWebhookAuthentication{Enabled: boolPointer(true)},
+		},
+		Authorization:         kubeletv1beta1.KubeletAuthorization{Mode: kubeletv1beta1.KubeletAuthorizationModeWebhook},
 		HealthzBindAddress:    "0.0.0.0",
-		OOMScoreAdj:           -1000,
+		OOMScoreAdj:           int32Pointer(-1000),
 		ClusterDomain:         domain,
-		RuntimeRequestTimeout: "15m",
-		FailSwapOn:            !allowSwap,
+		RuntimeRequestTimeout: metav1.Duration{Duration: 15 * time.Minute},
+		FailSwapOn:            boolPointer(!allowSwap),
 		ContainerLogMaxSize:   logSize,
-		ContainerLogMaxFiles:  logFiles,
+		ContainerLogMaxFiles:  int32Pointer(logFiles),
 	}
 }
 
-// KubeletAuthentication is a simplified version of the struct defined in
-// https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/apis/config/types.go
-//
-// Rationate: kubernetes repository is too large and not intended for client usage.
-type KubeletAuthentication struct {
-	ClientCAFile string
+func int32Pointer(input int32) *int32 {
+	return &input
 }
 
-// MarshalYAML implements yaml.Marshaler.
-func (a KubeletAuthentication) MarshalYAML() (interface{}, error) {
-	v := map[string]map[string]interface{}{}
-	v["x509"] = map[string]interface{}{"clientCAFile": a.ClientCAFile}
-	v["webhook"] = map[string]interface{}{"enabled": true}
-	v["anonymous"] = map[string]interface{}{"enabled": false}
-	return v, nil
+func boolPointer(input bool) *bool {
+	return &input
 }
 
-// MarshalJSON implements json.Marshaler.
-func (a KubeletAuthentication) MarshalJSON() ([]byte, error) {
-	v, err := a.MarshalYAML()
-	if err != nil {
-		return nil, err
+func newEncryptionConfiguration() apiserverv1.EncryptionConfiguration {
+	return apiserverv1.EncryptionConfiguration{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "apiserver.config.k8s.io/v1",
```

Comment:
Use predefined constants like this:
https://github.com/kubernetes/apiserver/blob/master/pkg/apis/config/v1/register.go#L25

Reply:
Or, it'd be better to use `scheme` and register the types to the scheme,
then encode the structs into JSON using `scheme`.